data-prepper-core/src/test/java/org/opensearch/dataprepper/parser/config/CustomTagsMeterFilterTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.parser.config;
+
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.opensearch.dataprepper.DataPrepper.getServiceNameForMetrics;
+import static org.opensearch.dataprepper.metrics.MetricNames.SERVICE_NAME;
+
+class CustomTagsMeterFilterTest {
+    private static final String testTagKey = "testKey";
+    private final String testFilterKey = "testFilterKey";
+    private final String testTagValue = "testValue";
+    private final String testFilterValue = "testFilterValue";
+    private final String testMetricName = "test-pipeline.grok.matchErrors.count";
+
+    private CustomTagsMeterFilter createObjectUnderTest(final String regexPattern) {
+        final MetricTagFilter metricTagFilter = new MetricTagFilter(regexPattern, Map.of(testFilterKey, testFilterValue));
+        return new CustomTagsMeterFilter(Map.of(testTagKey, testTagValue), List.of(metricTagFilter));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"test-pipeline.grok.**", "**.grok.**", "**.*Errors.*", "**.count", "**"})
+    void testMapShouldAddTagsIfRegexMatches(final String regex) {
+        final CustomTagsMeterFilter objectUnderTest = createObjectUnderTest(regex);
+
+        final Meter.Id testMeterId = new Meter.Id(testMetricName, Tags.empty(), null, null, Meter.Type.COUNTER);
+        final Meter.Id testMeterIdWithTags = objectUnderTest.map(testMeterId);
+
+        final List<Tag> expectedTags = List.of(Tag.of(testFilterKey, testFilterValue));
+
+        assertThat(testMeterIdWithTags, notNullValue());
+        System.out.println(testMeterIdWithTags);
```

Comment:
Please remove `System.out.println`

You can also reword your assertions to get better print outs when it fails to pass:

```
assertThat(testMeterIdWithTags.getTags(), allOf(
                     hasSize(1),
                     equalTo(expectedTags)
             ));
```

It will show you the tags (only the `.getTags()` result) when it fails.